aws/resource_aws_instance.go

```diff
@@ -1011,6 +1010,58 @@ func resourceAwsInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if d.HasChange("root_block_device") {
+		if rbd, ok := d.GetOk("root_block_device"); ok {
+			rbdL := rbd.([]interface{})
+			if len(rbdL) > 1 {
+				return fmt.Errorf("Cannot specify more than one root_block_device.")
+			}
+			for _, r := range rbdL {
+				bd := r.(map[string]interface{})
+				if volumeSize, ok := bd["volume_size"].(int); ok && volumeSize != 0 {
+					volumeIds, err := getAwsInstanceVolumeIds(conn, d)
```

Comment:
Do we know for sure that the first volume in the list is always RBD? The docs don't make it very clear 🤔  http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeVolumes.html and we'd rather not resize wrong volume by accident.